torch/__init__.py

```diff
@@ -1505,6 +1499,14 @@ def apply_mode(self, mode: Optional[str]):
         if mode is None or mode == "default":
             pass
         elif mode in ("reduce-overhead", "max-autotune", "max-autotune-no-cudagraphs"):
+            # This is a hacky place for this. If we need more of this, we should ostensibly either combine
+            # the configs to make this easier, or give the dynamo config an equivalent to_dict() and list_mode_options()
+            # or split list_mode_options into another key of "dynamo"/"inductor"
+            if mode == "reduce-overhead":
```

Comment:
Should we also do this on setattr of `cudagraphs` ? many users will just manipulate the config themselves instead of passing this mode in

Reply:
Ed's suggestion was based on mode, if we do this, then we might as well leave it as it is entirely right?

Reply:
Actually I like Elias's suggestion. Essentially, the change we should make is that when we consult automatic_dynamic, we should also consult triton.cudagraphs. If triton.cudagraphs is true, this forces automatic_dynamic to False (in fact, it is a little stronger than that: if triton.cudagraphs is true, we must never make things dynamic, even if the user asked us to do so!)

Reply:
It's a bit annoying that we can't delete this config option anymore but it seems exactly what you want to trigger the cudagraphs behavior.